mesonbuild/dependencies/base.py

```diff
@@ -347,32 +339,32 @@ def get_partial_dependency(self, *, compile_args: bool = False,
         if not sources:
             new.sources = []
         if not includes:
-            new.include_directories = []
+            pass # TODO maybe filter compile_args?
```

Comment:
I have a patch that fixes this for pkg-config, it's really cmake that I'm not sure how to fix... maybe you know how :)

Reply:
Just filtering for `self.compile_args` for anything starting with `-I` (or `-isystem`) would also work for CMake (not sure how you are implementing it for pkg-config). The CMake dependency class does detect the include directories separately from the other compiler arguments. However, it then just adds them to `compile_args` and forgets about them.

Ideally, we should separate the include directories out from the normal `compile_args`, but this would be another big refactoring...